java/README.md

```diff
@@ -1,311 +1,98 @@
-4. Click OK and wait for the project to open and build
-
-To build the tests against the Ice binary distribution you must set `ICE_BIN_DIST` environment
-variable to `all` before starting Android Studio.
-
-### Running the Android Test Suite
-
-The Android Studio project contains a `controller` app for the Ice test
-suite. Prior to running the app, you must disable Android Studio's Instant Run
-feature, located in File / Settings / Build, Execution, Deployment /
-Instant Run.
-
-Tests are started from the dev machine using the `allTests.py` script, similar
-to the other language mappings. The script uses Ice for Python to communicate
-with the Android app, therefore you must build the [Python mapping]
-(../python) before continuing.
-
-You also need to add the `tools\bin`, `platform-tools` and `emulator`
-directories from the Android SDK to your PATH. On macOS, you can use the
-following commands:
-
-```
-export PATH=~/Library/Android/sdk/cmdline-tools/latest/bin:$PATH
-export PATH=~/Library/Android/sdk/platform-tools:$PATH
-export PATH=~/Library/Android/sdk/emulator:$PATH
-```
-
-On Windows, you can use the following commands:
-
-```
-set PATH=%LOCALAPPDATA%\Android\sdk\cmdline-tools\latest\bin;%PATH%
-set PATH=%LOCALAPPDATA%\Android\sdk\platform-tools;%PATH%
-set PATH=%LOCALAPPDATA%\Android\sdk\emulator;%PATH%
-```
-
-Run the tests with the Android emulator by running the following command:
-
-```
-python allTests.py --android --controller-app
-```
-
-To run the tests on a Android device connected through USB, you can use
-the `--device=usb` option as shown below:
-
-```
-python allTests.py --android --device=usb --controller-app
-```
-
-To connect to an Android device that is running adb you can use the
-`--device=<ip-address>`
-
-```
-python allTests.py --android --device=<ip-address> --controller-app
-```
-
-To run the tests against a `controller` application started from Android
-Studio you should omit the `--controller-app` option from the commands above.
-
-## IceGrid GUI Tool
-
-Ice for Java includes the IceGrid GUI tool. It can be found in the file
-`lib/icegridgui.jar`.
-
-This JAR file is completely self-contained and has no external dependencies.
-You can start the tool with the following command:
-```
-java -jar icegridgui.jar
-```
-
-On macOS, the build also creates an application bundle named IceGrid GUI. You
-can start the IceGrid GUI tool by double-clicking the IceGrid GUI icon in
-Finder.
-
-[1]: https://zeroc.com/downloads/ice
-[2]: https://doc.zeroc.com/ice/3.7/release-notes/supported-platforms-for-ice-3-7-100
-[3]: https://gradle.org
-[4]: http://proguard.sourceforge.net
-[5]: https://commons.apache.org/proper/commons-compress/
+# Ice for Java
+
+[Getting started] | [Examples] | [Maven packages] | [Documentation] | [Building from source]
+
+The [Ice framework] provides everything you need to build networked applications, including RPC, pub/sub, server deployment, and more.
+
+Ice for Java is the Ice Java implementation.
```

Comment:
?

Should it be?
Ice for Java is the Java implementation of the Ice framework. It includes the latest Slice-to-Java mapping introduced in Ice X.Y.

Reply:
It was introduced in Ice 3.7.

Reply:
Fixed as suggested